modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GML2EncodingUtils.java

```diff
@@ -266,5 +269,42 @@ public static List GeometryPropertyType_getProperties(Geometry geometry) {
         return e.GeometryPropertyType_getProperties(geometry);
     }
     
+    /**
+     * Returns the geometry dimension, either as forced in the configuration, or the geometry
+     * natural one
+     * 
+     * @param geometry
+     * @param config
+     * @return
+     */
+    public static Integer getGeometryDimension(Geometry geometry, Configuration config) {
+        if (GMLEncodingUtils.isEmpty(geometry)) {
+            return null;
+        }
+
+        // check if srsDimension is turned off
+        if (config.hasProperty(GMLConfiguration.NO_SRS_DIMENSION)) {
+            return null;
+        }
+
+        /**
+         * For the dimension, use the actual dimension of the geometry. Using the dimension of the
+         * CRS is not sufficient, since currently CRSes don't support 3D.
+         */
+        return CoordinateSequences.coordinateDimension(geometry);
+    }
+
+    public static Integer getEnvelopeDimension(ReferencedEnvelope e, Configuration configuration) {
+        if (e == null || e.isNull() || e.getCoordinateReferenceSystem() == null) {
+            return null;
+        }
+
+        // check if srsDimension is turned off
+        if (configuration.hasProperty(GMLConfiguration.NO_SRS_DIMENSION)) {
+            return null;
+        }
+
+        return e.getCoordinateReferenceSystem().getCoordinateSystem().getDimension();
```

Comment:
CRS does not know about ReferencedEnvelope (but we could use opengis Envelope), more importantly, does not know about Configuration (a gt-xsd class)